### PR TITLE
Add `KaiserWindowState` bloq to prepare initial state for high confidence QPE

### PR DESCRIPTION
**Pull request by tanujkhattar**
Here is a nice plot (generated from the notebook added) that shows the comparison between using different window functions for QPE

image

### CODE REVIEW

File: qualtran/bloqs/phase_estimation/kaiser_window_state.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+        return [
+            qtn.Tensor(
+                data=self.kaiser_state_coeff.reshape((2,) * self.bitsize),
+                inds=[(outgoing['qpe_reg'], i) for i in range(int(self.bitsize))],
+                tags=[str(self)],
+            )
+        ]
+
+    def _kraus_(self) -> Iterable[np.ndarray]:
```
Comment (fdmalone) [inline qualtran/bloqs/phase_estimation/kaiser_window_state.py:120]: why is this added? do we actually test it anywhere?

**Review by fdmalone — APPROVED** [review]

File: qualtran/bloqs/phase_estimation/kaiser_window_state.py (hunk 1 above)
Comment (tanujkhattar) [reply qualtran/bloqs/phase_estimation/kaiser_window_state.py:120]: The notebook uses `cirq.Simulator()` to sample from the resulting circuit. To enable the sampling, we need this.

Comment (tanujkhattar) [reply qualtran/bloqs/phase_estimation/kaiser_window_state.py]: Refactored to do simulations using Qualtran and not rely on cirq simulator. Now inheriting from `Bloq` instead of `GateWithRegisters`.

File: qualtran/bloqs/phase_estimation/kaiser_window_state.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    where $M = 2^{m-1}$. See Ref[1] for more details.
+
+
+    Args:
+        bitsize: Number of bits in the control register of QPE.
+        alpha: Shape parameter, determines trade-off between main-lobe width and side lobe level.
+
+    References:
+        [Analyzing Prospects for Quantum Advantage in Topological Data
+        Analysis](https://arxiv.org/abs/2209.13581), Appendix D
```
[the remark this answers is not in the document]
Comment (tanujkhattar) [reply qualtran/bloqs/phase_estimation/kaiser_window_state.py]: done
